src/components/Post.jsx

```diff
@@ -1,85 +1,112 @@
-                {post.photos.map((photo, index) => (
-                  <img
-                    key={index}
-                    src={photo}
-                    alt={`${post.title} - ${index + 1}`}
-                    onClick={() => setSelectedPhotoIndex(index)}
-                    className={`w-full h-20 object-cover rounded-lg cursor-pointer border ${
-                      photo === selectedPhoto
-                        ? "border-primary border-2"
-                        : "border-gray"
-                    }`}
-                  />
-                ))}
-              </div>
-            )}
+  useEffect(() => {
+    setCoords(null);
+    if (post.zip) {
+      getCoordinatesByZip(post.zip)
+        .then(setCoords)
+        .catch((err) => console.error("Geocoding error:", err));
+    }
+  }, [post.item_id, post.zip]);
+
+  return (
+    <div className="flex flex-col lg:flex-row gap-6 w-full">
+      <div className="w-full lg:w-1/2">
+        <div className="rounded-2xl overflow-hidden">
+          <img src={selectedPhoto} alt={post.title} className="w-full h-auto" />
+        </div>
+        {post.photos?.length > 1 && (
+          <div className="grid grid-cols-3 sm:grid-cols-4 gap-2 mt-4">
+            {post.photos.map((photo, index) => (
+              <img
+                key={index}
+                src={photo}
+                alt={`${post.title} - ${index + 1}`}
+                onClick={() => setSelectedPhotoIndex(index)}
+                className={`w-full h-20 object-cover rounded-lg cursor-pointer border ${
+                  photo === selectedPhoto
+                    ? "border-primary border-2"
+                    : "border-gray"
+                }`}
+              />
+            ))}
           </div>
-      
-          <div className="w-full lg:w-1/2 mt-8 lg:mt-0">
-            <h2 className="text-4xl font-extrabold font-montserrat text-primary mb-2">
-              {post.title}
-            </h2>
-            <p className="text-base text-gray mb-6">{post.category}</p>
-      
-            <div className="mb-6">
-              <h3 className="text-lg font-semibold">Details:</h3>
-              <p>{post.description}</p>
-            </div>
-      
-            <div className="mb-6">
-              <h3 className="text-lg font-semibold">Address to meet:</h3>
-              <p>{post.location}</p>
-              <div className="w-full h-52 bg-gray-light rounded-xl flex items-center justify-center text-gray text-sm mt-4">
-                Map will be displayed here
-              </div>
+        )}
+      </div>
+
+      <div className="w-full lg:w-1/2 mt-8 lg:mt-0">
+        <h2 className="text-4xl font-extrabold font-montserrat text-primary mb-2">
+          {post.title}
+        </h2>
+        <p className="text-base text-gray mb-6">{post.category}</p>
+
+        <div className="mb-6">
+          <h3 className="text-lg font-semibold">Address to meet:</h3>
+          <p>{post.zip}</p>
```

Comment:
Would be good to have more information here.

```suggestion
          <p>{post.province || post.city}, {post.state_code} {post.zip}</p>
```